sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/data/Solace.java

```diff
@@ -68,4 +68,307 @@ public String getName() {
       return name;
     }
   }
+  /** Represents a Solace destination type. */
+  public enum DestinationType {
+    TOPIC,
+    QUEUE,
+    UNKNOWN
+  }
+
+  /** Represents a Solace message destination (either a Topic or a Queue). */
+  @AutoValue
+  @DefaultSchema(AutoValueSchema.class)
+  public abstract static class Destination {
+    /**
+     * Gets the name of the destination.
+     *
+     * @return The destination name.
+     */
+    @SchemaFieldNumber("0")
+    public abstract String getName();
+
+    /**
+     * Gets the type of the destination (TOPIC, QUEUE or UNKNOWN).
+     *
+     * @return The destination type.
+     */
+    @SchemaFieldNumber("1")
+    public abstract DestinationType getType();
+
+    static Builder builder() {
+      return new AutoValue_Solace_Destination.Builder();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setName(String name);
+
+      abstract Builder setType(DestinationType type);
+
+      abstract Destination build();
+    }
+  }
+
+  /** Represents a Solace message record with its associated metadata. */
+  @AutoValue
+  @DefaultSchema(AutoValueSchema.class)
+  public abstract static class Record {
+    /**
+     * Gets the unique identifier of the message, a string for an application-specific message
+     * identifier.
+     *
+     * <p>Mapped from {@link BytesXMLMessage#getApplicationMessageId()}
+     *
+     * @return The message ID, or null if not available.
+     */
+    @SchemaFieldNumber("0")
+    public abstract @Nullable String getMessageId();
+
+    /**
+     * Gets the payload of the message as a ByteString.
+     *
+     * <p>Mapped from {@link BytesXMLMessage#getBytes()}
+     *
+     * @return The message payload.
+     */
+    @SchemaFieldNumber("1")
+    public abstract ByteString getPayload();
+    /**
+     * Gets the destination (topic or queue) to which the message was sent.
+     *
+     * <p>Mapped from {@link BytesXMLMessage#getDestination()}
+     *
+     * @return The destination, or null if not available.
+     */
+    @SchemaFieldNumber("2")
+    public abstract @Nullable Destination getDestination();
+
+    /**
+     * Gets the message expiration time in milliseconds since the Unix epoch.
+     *
+     * <p>A value of 0 indicates the message does not expire.
+     *
+     * <p>Mapped from {@link BytesXMLMessage#getExpiration()}
+     *
+     * @return The expiration timestamp.
+     */
+    @SchemaFieldNumber("3")
+    public abstract long getExpiration();
+
+    /**
+     * Gets the priority level of the message (0-255, higher is more important). -1 if not set.
```

Comment:
Where in the code are these constraints enforced? Iimplementing one's own Builder build method with `autoBuild();` would allow for these constraints, however when doing this, you'll get an error when deriving a Beam Schema/Row. In such situations, I've ended up using a Pojo instead of Autovalue, though I prefer AutoValue when instantiating my own classes. I'll support whatever you decide but thought it would be helpful to enforce this behavior somewhere and write a few unit tests to validate.

Reply:
These are not actually constrains imposed in the code in this connector. These are constrains from the respective methods of BytesXMLMessage (for example, [BytesXMLMessag#getPriority](https://docs.solace.com/API-Developer-Online-Ref-Documentation/java/com/solacesystems/jcsmp/XMLMessage.html#getPriority--)) and I copy pasted them as this class is basically read-only (at least in the read connector), and a user will not create it manually. 

This got me thinking, if these restrictions should be mentioned here, or should I only leave a references to the relevant documentation pages of the BytesXMLMessage class. What do you think?